### PR TITLE
Add NVBench Algorithm Benchmarking

### PR DESCRIPTION
**Pull request by annielytical**

### CODE REVIEW

**Comment by annielytical** [comment]
for testing:  
```
make bc_bench
make bfs_bench
make color_bench
make geo_bench
make hits_bench
make kcore_bench
make mst_bench
make ppr_bench
make pr_bench
make spgemm_bench
make spmv_bench
make sssp_bench

sudo ./bin/bc_bench -m ../datasets/chesapeake/chesapeake.mtx
sudo ./bin/bfs_bench -m ../datasets/chesapeake/chesapeake.mtx
sudo ./bin/color_bench -m ../datasets/chesapeake/chesapeake.mtx
sudo ./bin/geo_bench -m ../datasets/chesapeake/chesapeake.mtx -c ../datasets/geolocation/sample.labels
sudo ./bin/hits_bench -m ../datasets/chesapeake/chesapeake.mtx
sudo ./bin/kcore_bench -m ../datasets/chesapeake/chesapeake.mtx
sudo ./bin/mst_bench -m ../datasets/chesapeake/chesapeake.mtx
sudo ./bin/ppr_bench -m ../datasets/chesapeake/chesapeake.mtx
sudo ./bin/pr_bench -m ../datasets/chesapeake/chesapeake.mtx
sudo ./bin/spgemm_bench -a ../datasets/spgemm/a.mtx -b ../datasets/spgemm/b.mtx
sudo ./bin/spmv_bench -m ../datasets/chesapeake/chesapeake.mtx
sudo ./bin/sssp_bench -m ../datasets/chesapeake/chesapeake.mtx
```

**Comment by neoblizz** [comment]
> for testing:
> 
> ```
> make bc_bench
> make bfs_bench
> make color_bench
> make geo_bench
> make hits_bench
> make kcore_bench
> make mst_bench
> make ppr_bench
> make pr_bench
> make spgemm_bench
> make spmv_bench
> make sssp_bench
> 
> sudo ./bin/bc_bench -m ../datasets/chesapeake/chesapeake.mtx
> sudo ./bin/bfs_bench -m ../datasets/chesapeake/chesapeake.mtx
> sudo ./bin/color_bench -m ../datasets/chesapeake/chesapeake.mtx
> sudo ./bin/geo_bench -m ../datasets/chesapeake/chesapeake.mtx -c ../datasets/geolocation/sample.labels
> sudo ./bin/hits_bench -m ../datasets/chesapeake/chesapeake.mtx
> sudo ./bin/kcore_bench -m ../datasets/chesapeake/chesapeake.mtx
> sudo ./bin/mst_bench -m ../datasets/chesapeake/chesapeake.mtx
> sudo ./bin/ppr_bench -m ../datasets/chesapeake/chesapeake.mtx
> sudo ./bin/pr_bench -m ../datasets/chesapeake/chesapeake.mtx
> sudo ./bin/spgemm_bench -a ../datasets/spgemm/a.mtx -b ../datasets/spgemm/b.mtx
> sudo ./bin/spmv_bench -m ../datasets/chesapeake/chesapeake.mtx
> sudo ./bin/sssp_bench -m ../datasets/chesapeake/chesapeake.mtx
> ```

Can you please put together a script for this instead?

**Comment by annielytical** [comment]
I was using one but wasn't sure where to put it since obviously make needs to be run from the build directory.

For now I put **test_benchmarks.sh** in the benchmarks directory but said at the top to run it from build.

File: benchmarks/test_benchmarks.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+sudo ${BIN_DIR}/color_bench -m ${MATRIX_FILE} --json ${JSON_DIR}/color.json
+sudo ${BIN_DIR}/geo_bench -m ${MATRIX_FILE} -c ${COORDINATES_FILE} --json ${JSON_DIR}/geo.json
+sudo ${BIN_DIR}/hits_bench -m ${MATRIX_FILE} --json ${JSON_DIR}/hits.json
+sudo ${BIN_DIR}/kcore_bench -m ${MATRIX_FILE} --json ${JSON_DIR}/kcore.json
+sudo ${BIN_DIR}/mst_bench -m ${MATRIX_FILE} --json ${JSON_DIR}/mst.json
+sudo ${BIN_DIR}/ppr_bench -m ${MATRIX_FILE} --json ${JSON_DIR}/ppr.json
+sudo ${BIN_DIR}/pr_bench -m ${MATRIX_FILE} --json ${JSON_DIR}/pr.json
+sudo ${BIN_DIR}/spgemm_bench -a ${A_MATRIX} -b ${B_MATRIX} --json ${JSON_DIR}/spgemm.json
+sudo ${BIN_DIR}/spmv_bench -m ${MATRIX_FILE} --json ${JSON_DIR}/spmv.json
+sudo ${BIN_DIR}/sssp_bench -m ${MATRIX_FILE} --json ${JSON_DIR}/sssp.json
```
Comment (neoblizz) [inline benchmarks/test_benchmarks.sh:48]: Why do these need to run as `sudo`?

**Pull request merged**